### PR TITLE
Fixed vulnerabilities

### LINKED ISSUE #47: Please bump got to 12.1 to fix Vuln
plz bump your got package. 

Details on security issue here:
https://github.com/advisories/GHSA-pfrx-2q88-qq97

### PR DESCRIPTION
**Pull request by springcomp**
This PR fixes #47. by replacing [got](https://www.npmjs.com/package/got) by [ky](https://www.npmjs.com/package/ky) when issueing an HTTP `HEAD` request.

This PR also fixes eslint warnings.

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
 export class InvalidNameError extends Error {}
 
-export interface Options {
+export class Options {
```
Comment (sindresorhus) [inline index.d.ts:3]: This should be `type`

File: index.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import isUrl from 'is-url-superb';
-import got from 'got';
+import ky from 'ky';
 import isScoped from 'is-scoped';
-import registryUrl from 'registry-url';
+import registryUrl from 'registry-auth-token/registry-url.js';
```
Comment (sindresorhus) [inline index.js:4]: You should not access files inside a package. That's bounds to break at some point.

File: index.js [hunk 3]
```diff
@@ -38,26 +38,26 @@ const request = async (name, options) => {
...
-		// eslint-disable-next-line unicorn/prefer-ternary
+		let packageUrl = registryUrl + urlName.toLowerCase();
 		if (isOrganization) {
-			await got.head(npmOrganizationUrl + urlName.toLowerCase(), {timeout: 10000});
-		} else {
-			await got.head(registryUrl + urlName.toLowerCase(), {timeout: 10000, headers});
+			packageUrl = npmOrganizationUrl + urlName.toLowerCase();
 		}
 
+		// eslint-disable-next-line unicorn/numeric-separators-style
```
Comment (sindresorhus) [inline index.js:56]: The lint rule is there for a reason.

File: index.js [hunk 4]
```diff
@@ -38,26 +38,26 @@ const request = async (name, options) => {
...
 	const isScopedPackage = isScoped(name);
 	if (isScopedPackage) {
-		urlName = name.replace(/\//g, '%2f');
+		urlName = name.replaceAll('/', '%2f');
 	}
 
 	const authInfo = registryAuthToken(registryUrl, {recursive: true});
 	const headers = {};
-	if (authInfo) {
+	if (authInfo && !isOrganization) {
```
Comment (sindresorhus) [inline index.js:46]: This change needs an explanation

File: index.js [hunk 5]
```diff
@@ -38,26 +38,26 @@ const request = async (name, options) => {
...
-			await got.head(registryUrl + urlName.toLowerCase(), {timeout: 10000, headers});
+			packageUrl = npmOrganizationUrl + urlName.toLowerCase();
 		}
 
+		// eslint-disable-next-line unicorn/numeric-separators-style
+		await ky.head(packageUrl, {timeout: 10000, headers});
 		return false;
 	} catch (error) {
-		const {statusCode} = error.response || {};
+		const statusCode = (error.response || {status: 500}).status;
```
Comment (sindresorhus) [inline index.js:60]: This is incorrect. If `error.response` is defined, it returns the whole response as status code.

And it's not clear why we can assume `500` is correct as fallback.

File: index.test-d.ts [hunk 6]
```diff
@@ -3,13 +3,14 @@ import npmName, {npmNameMany, InvalidNameError} from './index.js';
...
-	registryUrl: 'https://registry.yarnpkg.com/'
+	registryUrl: 'https://registry.yarnpkg.com/',
 }));
 
 const manyResult = npmNameMany(['chalk', '@sindresorhus/is', 'abc123']);
 expectType<Promise<Map<'chalk' | '@sindresorhus/is' | 'abc123', boolean>>>(
-	manyResult
+	manyResult,
 );
+// eslint-disable-next-line unicorn/no-await-expression-member
```
Comment (sindresorhus) [inline index.test-d.ts:13]: Fix the lint issue.

**Review by sindresorhus — CHANGES_REQUESTED** [review]

File: index.js (hunk 3 above)
Comment (springcomp) [reply index.js:56]: numeric separators are not valid until Node `12.5`. If we relax the version constraint from `package.json` we can use numeric separators.

File: index.js (hunk 4 above)
Comment (springcomp) [reply index.js:46]: Previously, there was two different code paths for issueing HTTP `HEAD` requests based on the condition `isOrganization`. When that was the case, the `Authorization` header was not used.

By refactoring the code, this PR builds the requested `header` and then has only a single instruction for HTTP request.

File: index.js (hunk 5 above)
Comment (springcomp) [reply index.js:60]: I beg to disagree.

The `.status` should return the status code in all cases.
To me an exception from an HTTP response is translated to `500` status code in virtually all frameworks I have ever come across. 

This code is a fallback to be safe as there should always be a valid response, and thus the real status code anyway.

File: index.js (hunk 2 above)
Comment (springcomp) [reply index.js:4]: Better that than having an extra dependency.

File: index.js (hunk 5 above)
Comment (springcomp) [reply index.js:60]: Besides, the unit-tests prove that this code, indeed, works 😊

**Comment by sindresorhus** [comment]
You will need to target Node.js 18

File: index.js (hunk 3 above)
Comment (springcomp) [reply index.js]: Targetted Node `>18`.

**Comment by sindresorhus** [comment]
Thanks :)

**Pull request merged**